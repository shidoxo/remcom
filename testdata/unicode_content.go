package main

import "fmt"

func main() {
	// Unicode strings with comments
	greeting := "Hello 世界" // World in Chinese
	emoji := "😀🎉🚀"         // Emoji string

	// Mathematical symbols
	formula := "E = mc²" // Einstein's formula
	greek := "α β γ δ ε" // Greek letters

	// Different languages
	japanese := "こんにちは" // Hello in Japanese
	russian := "Привет" // Hello in Russian
	arabic := "مرحبا"   // Hello in Arabic

	// Unicode in raw strings
	rawUnicode := `
        Symbols: ∑ ∏ ∫ ∂
        Arrows: → ← ↑ ↓
        Currency: € £ ¥ ¢
    ` // Raw string with symbols

	// Unicode variable names (valid in Go)
	π := 3.14159 // Pi constant
	θ := 45.0    // Theta angle

	fmt.Println(greeting, emoji)           // Print greeting
	fmt.Println(formula, greek)            // Print formula
	fmt.Println(japanese, russian, arabic) // Print languages
	fmt.Println(rawUnicode)                // Print raw unicode
	fmt.Println(π, θ)                      // Print mathematical variables
}
