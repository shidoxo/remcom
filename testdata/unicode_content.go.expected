package main

import "fmt"

func main() {
	greeting := "Hello 世界"
	emoji := "😀🎉🚀"

	formula := "E = mc²"
	greek := "α β γ δ ε"

	japanese := "こんにちは"
	russian := "Привет"
	arabic := "مرحبا"

	rawUnicode := `
        Symbols: ∑ ∏ ∫ ∂
        Arrows: → ← ↑ ↓
        Currency: € £ ¥ ¢
    `

	π := 3.14159
	θ := 45.0

	fmt.Println(greeting, emoji)
	fmt.Println(formula, greek)
	fmt.Println(japanese, russian, arabic)
	fmt.Println(rawUnicode)
	fmt.Println(π, θ)
}
